### PR TITLE
Parser

### PR DESCRIPTION
**Pull request by sjatkins12**

### CODE REVIEW

File: input_parser/parser.c [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "ast.h"
+
+static inline __attribute__((always_inline)) void	*init_parser(void)
+{
+	t_parser	*parser;
+
+	if (!(parser = malloc(sizeof(t_ast))) || !ft_memset(parser, 0, sizeof(t_parser)))
```
Comment (suedadam) [inline input_parser/parser.c]: Remove ternary memset ; Our malloc guarantees zero'd out memory

File: input_parser/parser.c [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+			return (NULL);
+		if (ret == CONTINUE)
+			continue ;
+		else if (ret == BREAK)
+			break ;
+		input_str++;
+	}
+	if (add_token(par->current_token, &(par->current_type), par) == EXIT_FAILURE)
+		return (NULL);
+	print_toks(par->tokens, par->types);
```
Comment (suedadam) [inline input_parser/parser.c]: Remove debug statements

File: input_parser/strappend.c [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+char				*strappend(char **str, char c)
+{
+	int				length;
+
+	length = ft_strlen(*str);
+	if (!(*str = realloc(*str, length + 2)))
+		return (NULL);
+	*str = ft_strncat(*str, &c, 1);
+	(*str)[length + 1] = 0;
```
Comment (suedadam) [inline input_parser/strappend.c]: Change to `\0` instead of `0`

File: input_parser/add_token.c [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+/*   By: satkins <satkins@student.42.fr>            +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2018/04/03 10:38:47 by satkins           #+#    #+#             */
+/*   Updated: 2018/04/03 16:30:03 by satkins          ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "ast.h"
+
+int					add_token(char *curr_token, t_token_type *curr_type,
```
Comment (suedadam) [inline input_parser/add_token.c:15]: Remove all commented out lines and fix alignment

File: input_parser/add_token.c [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+	t_parser *par)
+{
+	int				i;
+
+	if (!curr_token || !curr_type || *curr_type == null)
+		return (EXIT_SUCCESS);
```
Comment (suedadam) [inline input_parser/add_token.c]: Could you elaborate on why this is `EXIT_SUCCESS` and not a `EXIT_FAILURE`?

File: input_parser/parser.c [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+		ret = UNUSED_CHAR;
+		if (!(ret = is_semi(par, *input_str)) ||
+			(ret == UNUSED_CHAR && !(ret = is_op(par, *input_str))) ||
+			(ret == UNUSED_CHAR && !(ret = is_quote(par, *input_str))) ||
+			(ret == UNUSED_CHAR && !(ret = is_start_op(par, *input_str))) ||
+			(ret == UNUSED_CHAR && !(ret = is_whitespc(par, &input_str))) ||
+			(ret == UNUSED_CHAR && !(ret = is_word(par, *input_str))) ||
+			(ret == UNUSED_CHAR && !(ret = is_comment(par, *input_str))) ||
+			(ret == UNUSED_CHAR && !(ret = start_word(par, *input_str))))
+			return (NULL);
```
Comment (suedadam) [inline input_parser/parser.c]: Free all memory upon returning `NULL` (In all return statements)

**Review by suedadam — CHANGES_REQUESTED** [review]
Remove your .gitmodules file as well as your test files and executables (clean up after yourself).

Please also delete your libft folder as in the restructuring we're going to be using the libft in `deps/` which will conflict. 
Please also clean up the files you're not using (such as `parser.h`).

**Comment by sjatkins12** [comment]
Making a new pull request now!

**Pull request closed without merging**